Generator/Sources/NeedleFramework/Parsing/FileFilters/BasicKeywordFilter.swift

```diff
@@ -14,17 +14,18 @@
 //  limitations under the License.
 //
 
-import Foundation
 import SourceParsingFramework
 
-/// A filter that performs checks if the file content contains any
-/// dependency protocol declarations.
-class DependencyProtocolFilter: KeywordRegexFilter {
-
-    /// Initializer.
-    ///
-    /// - parameter content: The content to be filtered.
+/// A filter that does a simple String contains check for Needle related protocols/classes
+/// in a file.
+class BasicKeywordFilter: FileFilter {
+    private let content: String
+    
     init(content: String) {
-        super.init(content: content, keyword: "Dependency", regex: Regex.foundationInheritanceRegex(forProtocol: "Dependency"))
+        self.content = content
+    }
+    
+    func filter() -> Bool {
+        return content.containsAny(in: [componentClassName, dependencyProtocolName, pluginExtensionProtocolName])
```

Comment:
What's the perf difference with this vs. regex? 

To gain more speed, you can directly look up (and filter) the inherited type name during parsing instead of filtering here; it calls contains* which can be slow.

Reply:
It's an 10~15% improvement vs regex filtering.
It's strictly better performance because each of the 6 `KeywordRegexFilter` was doing a `String.contains` check first, before starting Regex filtering.

I'll try your other suggestions and see if they make a difference. Thanks!

Reply:
I made the change to stored the array as a private global variable. Didn't observe a noticeable performance gain though, but I'm keeping your suggestion anyways since theoretically it makes sense.

I'll play with the String -> Data technique separately, since it'll have to happen in the Uber's SwiftCommon repo.

Reply:
Data conversion should give a noticeable improvement. Alternatively, you could also try letting files be parsed and then look up the keywords in the inheritance clause in visit(_:) functions (and return .skipChildren); this might still be faster than the contains check.

Reply:
I've measured the latter (turn off filtering and just let everything be parsed), and saw performance gotten worse. AST parsing all the files took longer time compared to contains check, and we can't optimize that by making the syntax tree walks faster.

If I have more time I'll look at the data conversion, and url filtering. Which are the remaining two major candidates for performance gain.

Reply:
> I've measured the latter (turn off filtering and just let everything be parsed), and saw performance gotten worse. 

Is this with the top level parsing only, i.e. visit CodeBlockItem (or Class) at the top which then returns .skipChildren, or parsing all including members?  By default, it visits all children recursively so you'll have to explicitly return .skipChildren.

Reply:
I'm referring to the first step, `SyntaxParser.parse()`, which took longer time to run when I measured.
These `visit` methods happen after that (when we do `SyntaxVisitor.walk()`), I think?